internal/services/signalr/signalr_service_custom_domain_resource.go

```diff
@@ -0,0 +1,231 @@
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SignalR.SignalRClient
+			id, err := signalr.ParseCustomDomainID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.CustomDomainsGet(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading signalR custom domain %s: %+v", id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: got nil model", *id)
+			}
+
+			signalrServiceId := signalr.NewSignalRID(id.SubscriptionId, id.ResourceGroupName, id.SignalRName).ID()
+
+			state := CustomDomainSignalrServiceModel{
+				Name:             id.CustomDomainName,
+				DomainName:       resp.Model.Properties.DomainName,
+				SignalRServiceId: signalrServiceId,
+			}
+
+			if resp.Model.Properties.CustomCertificate.Id != nil {
+				signalrCustomCertId, err := signalr.ParseCustomCertificateIDInsensitively(*resp.Model.Properties.CustomCertificate.Id)
+				if err != nil {
+					return fmt.Errorf("parsing signalr custom cert id for %s: %+v", id, err)
+				}
+				state.SignalrCustomCertificateId = signalrCustomCertId.ID()
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r CustomDomainSignalrServiceResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SignalR.SignalRClient
+
+			id, err := signalr.ParseCustomDomainID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			signalrId := signalr.NewSignalRID(id.SubscriptionId, id.ResourceGroupName, id.SignalRName)
+
+			if err := waitForSignalrServiceStatusToBeReady(ctx, metadata, signalrId); err != nil {
+				return fmt.Errorf("waiting for signalR service %s state to be ready error: %+v", signalrId, err)
+			}
+
+			if err := client.CustomDomainsDeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CustomDomainSignalrServiceResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return signalr.ValidateCustomDomainID
+}
+
+func waitForSignalrServiceStatusToBeReady(ctx context.Context, metadata sdk.ResourceMetaData, id signalr.SignalRId) error {
+	signalrClient := metadata.Client.SignalR.SignalRClient
+
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending: []string{
+			string(signalr.ProvisioningStateUpdating),
+			string(signalr.ProvisioningStateCreating),
+			string(signalr.ProvisioningStateMoving),
+			string(signalr.ProvisioningStateRunning),
+		},
+		Target:                    []string{string(signalr.ProvisioningStateSucceeded)},
+		Refresh:                   signalrServiceProvisioningStateRefreshFunc(ctx, signalrClient, id),
+		Timeout:                   5 * time.Minute,
```

Comment:
this means we'll only ever poll for 5 minutes, which doesn't match the timeout set for this resource, we'll want to instead use the timeout from the context, for example:

```suggestion
		Timeout:                   time.Until(deadline),
```